### PR TITLE
Href Interpreter: Type Field

### PR DESCRIPTION
**Pull request by blankse**
| Q             | A
| ------------- | ---
| Bug fix?      | no
| New feature?  | yes
| BC breaks?    | no
| Deprecations? | no
| Fixed tickets |

I need an interpreter to get assets by id and get objects with unknown class.
This is possible with this element interpreter.

### CODE REVIEW

**Comment by dpfaffenbauer** [comment]
this is very similar to the Href interpreter, I would rather extend that one. WDYT?

**Comment by blankse** [comment]
@dpfaffenbauer OK I merged the Element interpreter with the Href interpreter

File: src/ImportDefinitionsBundle/Interpreter/HrefInterpreter.php [hunk 1]
```diff
@@ -30,14 +31,20 @@ class HrefInterpreter implements InterpreterInterface, DataSetAwareInterface
      */
     public function interpret(Concrete $object, $value, Mapping $map, $data, DefinitionInterface $definition, $params, $configuration)
     {
+        $type = $configuration['type'];
```
Comment (dpfaffenbauer) [inline src/ImportDefinitionsBundle/Interpreter/HrefInterpreter.php]: can we default the type to 'object'?

File: src/ImportDefinitionsBundle/Interpreter/HrefInterpreter.php [hunk 2]
```diff
@@ -30,14 +31,20 @@ class HrefInterpreter implements InterpreterInterface, DataSetAwareInterface
...
+
+        if ($type && $value) {
+            return Service::getElementById($type, $value);
+        }
+
         $objectClass = $configuration['class'];
```
Comment (dpfaffenbauer) [inline src/ImportDefinitionsBundle/Interpreter/HrefInterpreter.php]: and only do this if the type is object?

Comment (blankse) [reply src/ImportDefinitionsBundle/Interpreter/HrefInterpreter.php]: If the type is object, the Service::getElementById handle it already.
So if the class is set, the Service:getElementById shouldn't handle it?

Comment (dpfaffenbauer) [reply src/ImportDefinitionsBundle/Interpreter/HrefInterpreter.php]: no, since it bypasses the class configuration

**Comment by blankse** [comment]
@dpfaffenbauer Is it so okay?

**Comment by dpfaffenbauer** [comment]
that is fine now thanks :)

**Pull request merged**